scripts/buildWebtypes.js

```diff
@@ -0,0 +1,235 @@
+    value: {
+      // Type checking does not work with template tagged literals
+      // Since this Lit binding should use an expression, just declare it as such
+      kind: 'expression',
+    },
+  }));
+
+  return {
+    name: elementAnalysis.tagname,
+    description: elementAnalysis.description,
+    // Declare as extension to plain web type, this also means we don't have to
+    // repeat the same stuff from the plain web-types.json again
+    extension: true,
+    // IntelliJ does not understand Lit template syntax, so
+    // effectively everything has to be declared as attribute
+    attributes: [...booleanAttributes, ...propertyAttributes, ...eventAttributes],
+  };
+}
+
+function createLitWebTypes(packageJson, packageElements) {
+  return {
+    $schema: 'https://json.schemastore.org/web-types',
+    name: packageJson.name,
+    version: packageJson.version,
+    'description-markup': 'markdown',
+    framework: 'lit',
+    'framework-config': {
+      'enable-when': {
+        'node-packages': ['lit'],
+      },
+    },
+    contributions: {
+      html: {
+        elements: packageElements.map(createLitElementDefinition),
+      },
+    },
+  };
+}
+
+/**
+ * Creates web-types definitions for all relevant packages exposing
+ * web-components. Only public components are considered.
+ * The definitions are split into two files, one containing  "plain" types
+ * for the web-component, including attributes, properties and
+ * events. The other file contains lit-specific bindings, to bind regular
+ * properties, boolean properties, and events, through their respective lit
+ * attribute syntax.
+ */
+function buildWebTypes() {
+  const packages = getRelevantPackages();
+  const analysis = loadAnalysis();
+
+  packages.forEach((packageName) => {
+    const packageJson = JSON.parse(fs.readFileSync(`./packages/${packageName}/package.json`, 'utf8'));
+    const packageElements = analysis.elements
+      .filter((el) => el.path.startsWith(`packages/${packageName}/`))
+      .filter((el) => el.privacy === 'public');
+
+    const plainWebTypes = createPlainWebTypes(packageJson, packageElements);
+    const plainWebTypesJson = JSON.stringify(plainWebTypes, null, 2);
+    fs.writeFileSync(path.join('.', 'packages', packageName, PLAIN_WEB_TYPES_FILE), plainWebTypesJson, 'utf8');
+
+    const litWebTypes = createLitWebTypes(packageJson, packageElements);
+    const litWebTypesJson = JSON.stringify(litWebTypes, null, 2);
+    fs.writeFileSync(path.join('.', 'packages', packageName, LIT_WEB_TYPES_FILE), litWebTypesJson, 'utf8');
+  });
+}
+
+/**
+ * Updates all relevant packages' package.json files to reference the
+ * generated web-types JSON files, and to include them for publishing.
+ * This is not run by default. Instead, it can be run on demand when adding
+ * new packages, and the resulting changes can be committed to the repo.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function modifyPackageJson() {
```

Comment:
Consider adding a flag e.g. `--modify-package` to run this function from the script if needed.

Reply:
Added support for a `--modify-packages` flag.